spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java

```diff
@@ -32,25 +35,46 @@
 @ConfigurationProperties("spring.cloud.gcp.config")
 public class GcpConfigProperties implements CredentialsSupplier {
 
-	/** Enables Spring Cloud GCP Config. */
+	/**
+	 *  Enables Spring Cloud GCP Config.
+	 */
 	private boolean enabled;
 
-	/** Name of the application. */
+	/**
+	 *  Name of the application.
+	 */
+	@Value("${spring.application.name:application}")
 	private String name;
 
-	/** Profile under which app is running (e.g., "prod", "dev", "test", etc.). */
-	private String profile = "default";
+	/**
+	 *  Comma-delimited string of profiles under which the app is running.
+	 */
+	private String profile;
```

Comment:
Is the intent to allow for this property to be overwritten, but it defaults to `spring.profiles.active`? If so, the comment should reflect that.

Reply:
Added clarification, though my preference is actually that this property never gets used with our property prefix (`spring.cloud.gcp.config...`). Although it's there, in case an override is necessary for any reason...